app/components/channel_modal.js

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component, PropTypes} from 'react';
+import {Animated, Dimensions} from 'react-native';
+
+const AnimatedView = Animated.View;
+const {width: deviceWidth, height: deviceHeight} = Dimensions.get('window');
+
+export default class ChannelModal extends Component {
+    static propTypes = {
+        children: PropTypes.node,
+        duration: PropTypes.number,
+        topOffset: PropTypes.number,
```

Comment:
@jarredwitt Shouldn't `topOffset` be required? Without it, our calculation of the height of the `AnimatedView` below will be broken.